src/components/Courses/Courses.jsx

```diff
@@ -5,32 +5,81 @@ import CourseCard from './components/CourseCard/CourseCard';
 import Button from '../../common/Button/Button';
 import EmptyCourseList from '../../components/EmptyCourseList/EmptyCourseList';
 
-import { buttonNames } from '../../constants';
-import { useState } from 'react';
-import { useNavigate } from 'react-router-dom';
+import { buttonNames, urls, labels } from '../../constants';
+import { performGetRequest } from '../../services';
+import { saveCoursesAction } from '../../store/courses/actions';
+import { saveAuthorAction } from '../../store/authors/actions';
 
-import { useMainContext } from '../../context/MainContext';
+import { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
 
 const Courses = () => {
-	const { allCourses, filterCourses } = useMainContext();
+	const allCourses = useSelector((state) => state.courses);
+	const allAuthors = useSelector((state) => state.authors);
+	const [filteredCourses, setFilteredCourses] = useState(allCourses);
```

Comment:
Instead of using state here, you can use **useMemo** for this purpose which depends on allCourses and searchText

Reply:
const filteredCourses = useMemo(() => {
		if (!searchText) {
			return allCourses;
		}

		return allCourses.filter((course) =>
			course.title.toLowerCase().includes(searchText.toLowerCase())
		);
	}, [allCourses, searchText]);
	
filteredCourses - will be automatically updated and we will get needed behavior, list of courses will be changed/filtered on entering some search text. But we have Search button and from what I got, we need to perform search on Search button click, not automatically on entering search text.